machine/state.go

```diff
@@ -25,7 +25,7 @@ const (
 type MachineState struct {
 	ID       string
 	PublicIP string
-	Metadata map[string]string
+	Metadata map[string]string `json:"-"`
```

Comment:
Changed the serialization code to load metadata separately and combine them at the end.